djAerolith/wordwalls/templates/wordwalls/stats.html

```diff
@@ -0,0 +1,253 @@
+
+</div>
+
+<p>
+
+<div id="div2">
+
+    <canvas id="timeChart" width="400" height="100"></canvas>
+
+</div>
+
+</p>
+
+<p>
+
+Lexicon:<br>
+<select class="form-control" id="lexicon">
+  <option value="1">CSW15</option>
+  <option value="4">OWL2</option>
+  <option value="6">CSW12</option>
+  <option value="7">America</option>
+  <option value="8">FISE09</option>
+</select>
+
+<p>
+
+Challenges:<br>
+<select class="form-control" id="challenge">
+  <option value="6">Today's 7's</option>
+  <option value="7">Today's 8's</option>
+</select>
+
+</p>
+
+<button type="button" class="btn btn-primary" id="dropdown-button">Submit</button>
+
+</p>
+
+<script>
+
+// These hold info for the charts
+var scoresByDate = {};
+var dateArray = [];
+var timeRemainingByDate = {};
+
+$( "#dropdown-button" ).click(function() {
+    $("#scoresChart").load();
+
+    // This is the info we want
+    var lexicon = $('#lexicon').val()
+    var challenge = $('#challenge').val()
+
+    // Call the API, receive JSON
+    $.ajax({
+          method: "GET",
+          url: "/wordwalls/stats/api/" + lexicon + "/" + challenge,
+          dataType: 'json'
+    })
+        .done(function(msg) {
+
+            // Parse the JSON for the info we want for the charts
+            for (i = 0; i < msg.length; i++) {
+                var x = i;
+                var value = msg[i];
+                var dateString = value.Date;
+                var dateYearOnly = dateString.slice(0, 4);
+                var scoreShortened = (Number(value.Score) / Number(value.maxScore)) * 100;
+
+                scoresByDate[value.Date] = scoreShortened.toFixed(2);
+                timeRemainingByDate[value.Date] = value.timeRemaining;
+
+                // dateArray.push(dateYearOnly);
+                dateArray.push(value.Date);
+
+            }
+
+            // Call the charts function to generate the charts on the page
+            charts();
+        });
+
+function isDateInArray(value, array) {
+    return array.indexOf(value) > -1;
+}
+
+function charts () {
+
+    // Info for chart axes
+    dateArray = dateArray.sort();
+    var keys = [];
+    scoresSorted = [];
+    timeSorted = [];
+
+    // for (i = 0; i < dateArray.length; i++) {
+    //     if (isDateInArray(dateArray[i], dateArray) == false) {
+    //         dateArray.push(dateYearOnly);
+    //     } else {
+    //         dateArray.push(' ');
+    //     }
+    // }
+    // console.log(dateArray);
+
+    // for (i = 1; i < dateArray.length; i++) {
+    //     if (isDateInArray(dateArray[i], dateArray) == true) {
+    //         dateArray[i] = ' ';
+    //     }
+    // }
+        // if (dateArray[i] === dateArray[i - 1] || dateArray[i] in dateArray) {
+        //     dateArray[i] = ' ';
+        // }
+    // }
+    // console.log(dateArray);
+
+    for (var key in scoresByDate) {
+        if (scoresByDate.hasOwnProperty(key)) {
+            keys.push(key);
+        }
+    }
+
+    keys.sort();
```

Comment:
we can probably remove this because you already have an array of dates in `dateArray`